Add Salt for configuration management (#703)
theforeman/foreman-documentation: guides/common/modules/proc_salt-salt-example.adoc

[id="salt_guide_salt_example_{context}"]
= Salt Example

This example uses a Salt State to manage the `/etc/motd` file on one or more Salt Minions.
It demonstrates the use of {Project} as an external node classifier and the use of Salt Grains.

.Procedure
. Create a global parameter called `vendor_name` with the string `{Project}` as its value.
. Add a new Salt State called `motd` to your Salt Master.
.. Create the `/srv/salt/motd/` directory.
.. Add two files to it:

> ... The `/srv/salt/motd/init.sls` file is a Salt State file:

Reviewer: We never go this deep. 
You know, I see no value in indenting substeps if the user has to do them sequentially. 
I would appreciate if there isn't three levels of indentation. I'd rather no sub steps, since it looks to me that the user has to work through these all, so the hierarchy follows the UI rather the user, and we are writing user-oriented docs. 
Up to you though.

Author: Good call, fixed.

+
[options="nowrap" subs="attributes"]
----
/etc/motd:
  file.managed:
    - user: root
    - group: root
    - mode: 0644
    - source: salt://motd/motd.template
    - template: jinja
----
... The `/srv/salt/motd/motd.template` file is a template referenced from your Salt State file:
+
[options="nowrap" subs="attributes"]
----
Welcome to {{ grains['fqdn'] }} Powered by {{ salt['pillar.get']('vendor_name') }}
----
+
Access the `fqdn` Salt Grain from within this template and retrieve the `vendor_name` parameter from the Salt Pillar.
. Import the `motd` {ManagingHostsDocURL}salt_guide_salt_states[Salt State] into {Project}.
. Verify that Salt has been given access to the `vendor_name` parameter by running either of the following commands on your Salt Master:
+
[options="nowrap" subs="attributes"]
----
# salt '*' pillar.items | grep -A 1 vendor_name
# salt '*' pillar.get vendor_name
----
+

> If the output does not include the value of the `vendor_name` parameter, you need to refresh the Salt Pillar data first:

Reviewer: ```suggestion
If the output does not include the value of the `vendor_name` parameter, you must refresh the Salt Pillar data first:
```

Author: Applied.

+
[options="nowrap" subs="attributes"]
----
# salt '*' saltutil.refresh_pillar
----
+

> Refer to {ManagingHostsDocURL}salt_guide_enc_parameters[Salt ENC parameters] on how to refresh the pillar data.

Reviewer: ```suggestion
For information about how to refresh pillar data, see to {ManagingHostsDocURL}salt_guide_enc_parameters[Salt ENC parameters].
```

Seems pillar isn't capitalised here and it is in other places, can you ctrl + f and make it one way or the other consistently?

Author: Applied and capitalized `Salt Pillar`.
